src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForTerms.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Client.Extensions;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Documents.Queries.Dynamic;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Handlers.Processors.Indexes
+{
+    internal class IndexHandlerProcessorForTerms : AbstractIndexHandlerProcessorForTerms<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        private readonly OperationCancelToken _token;
+        private readonly long? _existingResultEtag;
+
+        public IndexHandlerProcessorForTerms([NotNull] DatabaseRequestHandler requestHandler, OperationCancelToken token, long? existingResultEtag)
+            : base(requestHandler, requestHandler.ContextPool)
+        {
+            _token = token;
+            _existingResultEtag = existingResultEtag;
+        }
+
+        protected override ValueTask<TermsQueryResultServerSide> GetTerms(TransactionOperationContext context2, string indexName, string field, string fromValue, int pageSize)
+        {
+            using (_token)
```

Comment:
why don't you create that token here instead of passing it via ctor? Actually this token should be created in abstract and passed to GetTermsAsync.

Reply:
`CreateTimeLimitedOperationToken()` method is protected

Reply:
done